File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/mediator/solution/SetupClass.java
// SetupClass.java

public class SetupClass
{
	public static Consumer setup()
	{
		Producer aProducer1 = new Producer("Alpha");
		Producer aProducer2 = new Producer("Bravo");
		Producer aProducer3 = new Producer("Charlie");
		Producer aProducer4 = new Producer("Delta");
		Producer aProducer5 = new Producer("Echo");
		Producer aProducer6 = new Producer("Foxtrot");

		Middleman aMiddleman = new Middleman();
		aMiddleman.addProducer(aProducer1);
		aMiddleman.addProducer(aProducer2);
		aMiddleman.addProducer(aProducer3);
		aMiddleman.addProducer(aProducer4);
		aMiddleman.addProducer(aProducer5);
		aMiddleman.addProducer(aProducer6);

		Consumer aConsumer = new Consumer("Zulu", aMiddleman);

		return aConsumer;
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/mediator/solution/setup_class.cpp
// setup_class.cpp

#include "consumer.h"
#include "middleman.h"
#include "producer.h"
#include "setup_class.h"

consumer * setup_class::setup()
{
	producer * a_producer1 = new producer("Alpha");
	producer * a_producer2 = new producer("Bravo");
	producer * a_producer3 = new producer("Charlie");
	producer * a_producer4 = new producer("Delta");
	producer * a_producer5 = new producer("Echo");
	producer * a_producer6 = new producer("Foxtrot");

	middleman * a_middleman = new middleman();
	a_middleman->add_producer(a_producer1);
	a_middleman->add_producer(a_producer2);
	a_middleman->add_producer(a_producer3);
	a_middleman->add_producer(a_producer4);
	a_middleman->add_producer(a_producer5);
	a_middleman->add_producer(a_producer6);

	consumer * a_consumer = new consumer("Zulu", a_middleman);

	return a_consumer;
}
